packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -405,17 +426,37 @@ Future<void> _buildGradleProjectV2(
     command.add('-Ptarget-platform=${getNameForTargetPlatform(buildInfo.targetPlatform)}');
 
   command.add(assembleTask);
+  bool potentialAndroidXFailure = false;
   final int exitCode = await runCommandAndStreamOutput(
     command,
     workingDirectory: flutterProject.android.hostAppGradleRoot.path,
     allowReentrantFlutter: true,
     environment: _gradleEnv,
-    filter: logger.isVerbose ? null : ndkMessageFilter,
+    mapFunction: (String line) {
+      if (androidXFailureRegex.hasMatch(line)) {
+        potentialAndroidXFailure = true;
+      }
+      // Always print the full line in verbose mode.
+      if (logger.isVerbose) {
+        return line;
+      } else if (androidXPluginWarningRegex.hasMatch(line) || !ndkMessageFilter.hasMatch(line)) {
+        return null;
+      }
+
+      return line;
+    }
   );
   status.stop();
 
-  if (exitCode != 0)
+  if (exitCode != 0) {
+    if (potentialAndroidXFailure) {
+      printError('*******************************************************************************************');
+      printError('The Gradle failure may have been because of AndroidX incompatibilities in this Flutter app.');
+      printError('See [WIP link] for more information on the problem and how to fix it.');
```

Comment:
We should link to a wiki page or something that we can update if the situation changes

Reply:
It's linking to flutter/website#2349.